SIS_import/sis_set_url.rb

```diff
@@ -24,15 +73,134 @@ def refreshESBToken()
 end
 
 ## make Canvas API call
-def Canvas_API_call(url, authorization_string)
+def Canvas_API_call(url, params, authorization_string, json_attribute, outputFile)
+	# make sure the call is within API usage quota
+	Canvas_sleep_according_to_timer_and_api_call_limit()
+
+	url = url<<"?"<<URI.encode_www_form(params)
+
+	response = actual_Canvas_API_call(url, authorization_string, outputFile)
+
+	json = parse_canvas_API_response_json(url, response, json_attribute, outputFile)
+
+	# array of page urls if any
+	page_urls = get_all_canvas_page_urls(response.headers)
+	if (!page_urls.nil?)
+		# there is paging involved
+		# need to make further API calls
+		# will concat the result json arrays
+		page_urls.each do |url|
+			response = actual_Canvas_API_call(url, authorization_string, outputFile)
+
+			json_paging_data = parse_canvas_API_response_json(url, response, json_attribute, outputFile)
+			# merge in all element in the paging data array
+			json_paging_data.each do |json_paging_data_element|
+				json.push json_paging_data_element
+			end
+		end
+	end
+	outputFile.write("Canvas API call JSON \n #{json}\n")
```

Comment:
commented out. Should be debug only.
As for the output file, there is one file generated for each script run, named after "Canvas_set_url_<timestamp>.txt"